lib/screens/downloads_settings_screen.dart

```diff
@@ -121,6 +120,35 @@ class RequireWifiSwitch extends StatelessWidget {
   }
 }
 
+class SyncFavoritesSwitch extends StatelessWidget {
+  const SyncFavoritesSwitch({super.key});
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder<Box<FinampSettings>>(
+      valueListenable: FinampSettingsHelper.finampSettingsListener,
+      builder: (context, box, child) {
+        bool? syncFavorites = box.get("FinampSettings")?.trackOfflineFavorites;
+
+        return SwitchListTile.adaptive(
+          title: Text(AppLocalizations.of(context)!.trackOfflineFavorites),
+          subtitle:
+              Text(AppLocalizations.of(context)!.trackOfflineFavoritesSubtitle),
+          value: syncFavorites ?? false,
+          onChanged: syncFavorites == null
+              ? null
+              : (value) {
+                  FinampSettings finampSettingsTemp =
+                      box.get("FinampSettings")!;
+                  finampSettingsTemp.trackOfflineFavorites = value;
+                  box.put("FinampSettings", finampSettingsTemp);
```

Comment:
I noticed you not using the `FinampSettingsHelper` recently. Is that on purpose? If so, I'm interested in your reasons.  
To my understanding the helper just makes it easier to change settings from anywhere in the app, and I don't think there's a downside to that, but I might be wrong of course.

Reply:
It just seems like a needless abstraction.  Why would we ever want to change this setting from outside this widget?  This is basically the same logic as why I prefer to not break the settings switch widgets out into separate files.  The breakout is especially pointless because most of the settings end up directly using the settings box anyway, so we don't even get to abstract that aspect away to a higher level interface with the submethod, we just spread our box manipulation into multiple methods for no reason.

Reply:
Hmm. I agree that for most settings it doesn't make a whole lot of sense. For some (like playback speed or potentially offline mode) it would make more sense, but I guess in that case we can just add a method to the helper.  
Would some abstraction that removes the need to always duplicate the box handling make sense? So that it's a generic `updateSetting(String settingKey, dynamic value)` for everything?

Reply:
Yeah, more abstractions in the settings would be nice to reduce box handling and/or widget duplication, but I haven't been able to think of a workable way to implement it.